src/Raven.Server/Documents/Queries/FieldsToFetch.cs

```diff
@@ -135,7 +135,15 @@ ref ignored
                 }
             }
 
-            var extract = indexDefinition.MapFields.TryGetValue(selectFieldName, out var value) && value.Storage == FieldStorage.Yes;
+            var key = selectFieldName.Value.Length == 0 &&
```

Comment:
Can we _try_ to put this condition into a method with some meaningful name? Something that would describe what happened that all those conditions are satisfied.

Reply:
done